.github/actions/fullnode-sync/fullnode_sync.py

```diff
@@ -52,6 +56,30 @@ def get_synced_version_from_index_response(api_index_response, exit_if_none):
   return int(synced_version)
 
 
+def get_metric_from_metrics_port(metric_name):
+  """Gets and returns the metric from the metrics port"""
+  # Ping the metrics port
+  process = subprocess.Popen(["curl", "-s", LOCAL_METRICS_ENDPOINT], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+  metrics_response, errors = process.communicate()
+  if metrics_response is None:
+    print_error_and_exit("Exiting! Unable to get the metrics from the localhost. Response is empty.")
+  if errors is not None and errors != b'':
+    print("Found output on stderr for get_synced_version_from_metrics_port: {errors}".format(errors=errors))
+
+  # Parse the metric value
+  try:
+    metrics_response = json.loads(metrics_response)
+    metric_value = metrics_response[metric_name]
+  except Exception as exception:
+    print("Exception caught when getting the metric value: {metric_name}. Exception: {exception}".format(metric_name=metric_name, exception=exception))
+    metric_value = 0
+  if metric_value is None:
+    print_error_and_exit("Exiting! Unable to get the metric from the metrics port. Metric is empty: {metric_name}".format(metric_name=metric_name))
```

Comment:
Do you need some sort of retry here?

Reply:
We're only ever pinging the metrics on the localhost, and I've yet to see that fail, i.e., despite having 8 tests, each run 5+ times for longer than several hours and calling this function every few seconds, it's never failed 😄 Of course, we could add a retry and a max retry counter. But, I'd rather wait until we get there 😄